src/querying_subscriber.rs

```diff
@@ -203,13 +185,11 @@ pub unsafe extern "C" fn ze_declare_querying_subscriber(
                 }
             }
             match sub
-                .callback(move |sample| {
-                    let payload = sample.payload.contiguous();
-                    let owner = match payload {
-                        std::borrow::Cow::Owned(v) => zenoh::buffers::ZBuf::from(v),
-                        _ => sample.payload.clone(),
-                    };
-                    let sample = z_sample_t::new(&sample, &owner);
+                .callback(move |mut sample| {
+                    if let std::borrow::Cow::Owned(v) = sample.payload.contiguous() {
```

Comment:
I suppose since we have api for working with non-contiguous ZBuf, we no longer need to convert payload to contiguous.

Reply:
Correct :)